app/src/main/java/com/SecUpwN/AIMSICD/smsdetection/SmsDetector.java

```diff
@@ -122,372 +116,187 @@ public void run() {
-                                );*/
-                                dbacess.toEventLog(4,"Detected MWI SMS");
-                                MiscUtils.startPopUpInfo(tContext, 7);
-                            } else {
-                                Log.d(TAG, " Detected Sms already logged");
-                            }
+        CapturedSmsData setmsg = new CapturedSmsData();
+        String smstext = findSmsData(bufflines,null);
+        String num = findSmsNumber(bufflines,null);
 
-                        } else if("WAPPUSH".equals(SILENT_ONLY_TAGS[index].split("#")[1].trim())) {
-                            /*
-                                Wap Push in logcat shows no data only senders number
-                                TODO: data is probably in db content://raw/1?
-                             */
-                            CapturedSmsData setmsg = new CapturedSmsData();
-                            setmsg.setSenderNumber("unknown");
-                            setmsg.setSenderMsg("no data");
-
-                            int startindex = x - 2;
-                            int endindex = x + 3;
-                            /*  wap push port DestPort 0x0B84
-                             *  its usually at this index of +3 in array                             *
-                              * */
-                            //This is index on Samsungs is different for other phone makes
-                            if (progress[x + 3].contains("DestPort 0x0B84")) {
-                                Log.d(TAG, "WAPPUSH DETECTED");
-                                /* loop thru array to find number */
-                                if (endindex+3 <= progress.length) {
-                                    while (startindex < endindex)
-                                    {
-                                        if (progress[startindex].contains(DETECTION_PHONENUM_SMS_DATA[2].toString())) {
-                                            try {
-                                                // Looking for OrigAddr this is where sender number is
-                                                String number = progress[startindex].substring(
-                                                        progress[startindex].indexOf("OrigAddr")).replace(
-                                                        DETECTION_PHONENUM_SMS_DATA[2].toString(), "").trim();
-
-                                                setmsg.setSenderNumber(number);
-                                                break;
-                                            } catch (Exception ee) {
-                                                Log.e(TAG, "Error parsing number", ee);
-                                            }
-
-                                        }
-                                        startindex++;
-
-                                    }
-                                }
+        if(smstext == null){smstext = "null";}
```

Comment:
It doesn't make a difference to compiler and slims down code rather than whitespace